### PR TITLE
Package exports for Vercel Edge Runtime support

### PR DESCRIPTION
**Pull request by PonomareVlad**
### Details

**Vercel Edge Runtime** need a different key in `"exports"` for **web** target

If we add a `"browser"` target, **Vercel Edge Runtime** can automatically switch to **web** build instead specifying it manually:

```js
import { Bot } from "grammy/web"; // now

import { Bot } from "grammy"; // after merge
```

Also, this can solve incompatibility for some plugins that not depend on **Node** internal packages, but uses default import for core **grammY** package

### Might be able to fix

- [ ] https://github.com/grammyjs/grammY/issues/372
- [ ] https://github.com/grammyjs/grammY/issues/435
- [ ] https://github.com/vercel/next.js/issues/51313

### How to test

I think we need to test this change on other platforms before merging 🌚

```json
{
  "dependencies": {
    "grammy": "npm:@ponomarevlad/grammy@1.17.0-vercel-edge"
  }
  "overrides": {
    "grammy": "npm:@ponomarevlad/grammy@1.17.0-vercel-edge"
  }
}
```

### Where need to test

- [x] [Node](https://github.com/PonomareVlad/grammYExportsTest/tree/node)
- [x] [Deno](https://github.com/PonomareVlad/grammYExportsTest/tree/deno)
- [x] ~~Deno Deploy~~ (same as Deno)
- [x] [Vercel Serverless](https://github.com/PonomareVlad/grammYExportsTest/tree/vercel-serverless)
- [x] [Vercel Edge Runtime](https://github.com/PonomareVlad/grammYExportsTest/tree/vercel-edge)
- [x] ~~Supabase Edge Functions~~ (same as Deno)
- [x] [Google Cloud Functions](https://github.com/PonomareVlad/grammYExportsTest/tree/google-cloud-functions)
- [x] [Cloudflare Workers](https://github.com/PonomareVlad/grammYExportsTest/tree/cloudflare)
- [x] [Firebase Functions](https://github.com/PonomareVlad/grammYExportsTest/tree/firebase-functions)
- [x] [Netlify Functions](https://github.com/PonomareVlad/grammYExportsTest/tree/netlify)

> [Complete example for Vercel Edge Runtime](https://github.com/PonomareVlad/grammYVercelEdge/tree/main)

### CODE REVIEW

**Review by KnorpelSenf — COMMENTED** [review]
Why isn't this covered by `default`? 🫠

**Comment by PonomareVlad** [comment]
> Why isn't this covered by `default`? 🫠

@balazsorban44 Can you comment on this question or suggest someone from @Vercel ?

File: src/composer.ts [hunk 1]
```diff
@@ -332,11 +332,11 @@ export class Composer<C extends Context> implements MiddlewareObj<C> {
...
-     * > (https://core.telegram.org/bots/features#deep-linking) to let users
-     * > start your bot with a custom payload. As an example, send someone the
-     * > link https://t.me/name-of-your-bot?start=custom-payload and register a
-     * > start command handler on your bot with grammY. As soon as the user
-     * > starts your bot, you will receive `custom-payload` in the `ctx.match`
+     * > (https://core.telegram.org/bots/features#deep-linking) to let users start your
+     * > bot with a custom payload. As an example, send someone the link
+     * > https://t.me/name-of-your-bot?start=custom-payload and register a start
+     * > command handler on your bot with grammY. As soon as the user starts
+     * > your bot, you will receive `custom-payload` in the `ctx.match`
```
Comment (KnorpelSenf) [inline src/composer.ts:339]: Please revert

**Review by KnorpelSenf — APPROVED** [review]
LGTM, thank you!

Also, please accept the invitation to join @grammyjs.

**Comment by KnorpelSenf** [comment]
Consider signing your commits in the future. We usually require this, but I used admin rights to merge anyway now.

**Comment by KnorpelSenf** [comment]
@all-contributors add @PonomareVlad for the extensive testing and the code to bring grammY to a new platform

**Pull request merged**